### PR TITLE
update libp2p 0.43.0

### PR DESCRIPTION
**Pull request by wngr**

### CODE REVIEW

**Review by dvc94ch — APPROVED** [review]

**Review by rklaehn — APPROVED** [review]

File: .gitignore [hunk 1]
```diff
@@ -1 +1,2 @@
 /target
+Cargo.lock
```
Comment (rkuhn) [inline .gitignore:2]: Why? I thought it is essential to record changes to this file.

File: src/lib.rs [hunk 2]
```diff
@@ -138,6 +125,41 @@ impl NetworkBehaviour for Broadcast {
...
+
+    fn inject_connection_established(
+        &mut self,
+        peer: &PeerId,
+        _connection_id: &ConnectionId,
+        _endpoint: &libp2p::core::ConnectedPoint,
+        _failed_addresses: Option<&Vec<Multiaddr>>,
+        _other_established: usize,
+    ) {
+        self.inject_connected(peer)
```
Comment (rkuhn) [inline src/lib.rs:150]: This is not correct: needs to check `other_established == 0` (similar below).

**Review by rkuhn — COMMENTED** [review]
I’ll open a new PR with the fixes.

File: .gitignore (hunk 1 above)
Comment (wngr) [reply .gitignore:2]: https://doc.rust-lang.org/cargo/faq.html#why-do-binaries-have-cargolock-in-version-control-but-not-libraries

**Comment by wngr** [comment]
Thanks!

**Pull request merged**